libcxx/test/benchmarks/format/write_string_comparison.bench.cpp

```diff
@@ -0,0 +1,220 @@
+                    }));                                                                                               \
+  BENCHMARK_CAPTURE(BM_format_to_iterator, <std::string> name, variable, ([] {                                         \
+                      std::string s;                                                                                   \
+                      s.resize(10'000);                                                                                \
+                      return s;                                                                                        \
+                    }));                                                                                               \
+  BENCHMARK_CAPTURE(BM_format_to_iterator, <std::vector> name, variable, ([] {                                         \
+                      std::vector<char> v;                                                                             \
+                      v.resize(10'000);                                                                                \
+                      return v;                                                                                        \
+                    }));                                                                                               \
+  BENCHMARK_CAPTURE(BM_format_to_iterator, <std::deque> name, variable, ([] {                                          \
+                      std::deque<char> d;                                                                              \
+                      d.resize(10'000);                                                                                \
+                      return d;                                                                                        \
+                    }));                                                                                               \
+                                                                                                                       \
+  /* */
+
+// Verify these types have an iterator that format has optimizations for.
+LIBCPP_STATIC_ASSERT(std::same_as<std::array<char, 1>::iterator, char*> || // the type depends on an ABI flag
+                     std::same_as<std::array<char, 1>::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(std::same_as<std::string::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(std::same_as<std::vector<char>::iterator, std::__wrap_iter<char*>>);
+
+// Verify these types have an iterator that format does not optimize for
+LIBCPP_STATIC_ASSERT(!std::same_as<std::deque<char>::iterator, char*> &&
+                     !std::same_as<std::deque<char>::iterator, std::__wrap_iter<char*>>);
+LIBCPP_STATIC_ASSERT(!std::same_as<std::list<char>::iterator, char*> &&
+                     !std::same_as<std::list<char>::iterator, std::__wrap_iter<char*>>);
```

Comment:
I don't think these checks belong in this benchmark. I would add them to a libc++ specific test file for `<format>`, maybe something like `libcxx/test/libcxx/<...>/format-optimizations.compile.pass.cpp`. The goal of that test would be to ensure that we don't regress these optimizations.